ckanext/dia/commands.py

```diff
@@ -69,22 +57,55 @@ def cleanup_datastore_once(self, start_offset):
             sys.exit(1)
 
         # query datastore to get all resources from the _table_metadata
-        resource_id_list = []
-        total_tested = 0
-        for offset in itertools.count(start=start_offset, step=100):
-            print "Load metadata records from datastore (offset: %s)" % offset
-            record_list, tested = self._get_datastore_table_page(context, offset)  # noqa
-            total_tested += tested
-            resource_id_list.extend(record_list)
-            if not tested > 0:
-                break
-            if len(resource_id_list) > 250:
+        print("Loading metadata records from datastore")
+        all_datastore_tables = self._get_all_datastore_tables(context)
```

Comment:
Get all the records in one call, then iterate over chunks below.
Each datastore table name in a chunk is then tested to see if there is a corresponding resource in the resource table, and it is marked as an orphan if none is found.
After building a list of orphaned datastore tables for the current chunk, they are deleted and count of success/failures tallied.
When all chunks are done, report.

Reply:
The actual intent of the script hasn't really changed from the original, just the way that the dataset is retrieved and iterated over.